### ISSUE TITLE
System.Runtime.CompilerServices.Unsafe

### ISSUE THREAD

**Issue by ajw333** [dotnet/sdk#24112]
Failing with the following:

> System.TypeLoadException: Could not load type 'System.Runtime.CompilerServices.Unsafe' from assembly 'System.Runtime, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.

This PR is probably blocked on #24009, which will update the version of the runtime used and should fix this error.

_Originally posted by @dsplaisted in https://github.com/dotnet/sdk/pull/24039#issuecomment-1050893761_

**Comment by dsplaisted** [comment]
I think this was fixed

**Comment by Alerinos** [comment]
![image](https://user-images.githubusercontent.com/42449535/163502034-86a162f6-f5a2-48b9-a32b-3f3b2addc662.png)
@dsplaisted  I have the same problem after updating .net core 7 preview 3

**Comment by dsplaisted** [comment]
@Alerinos Could you open a new issue, with some more information to help us reproduce the issue (ideally a repro project)?

**Comment by Alerinos** [comment]
@dsplaisted Thank you for writing back, I fixed the problem. It turned out that updating visual studio to the latest version did not update my .net, I had to manually download SDK + ASP.Net
